### ISSUE TITLE
Remove binary support

### ISSUE THREAD

**Issue by jagerber48** [jagerber48/sciform#201]
Currently `sciform` supports formatting numbers so that they have a base 2 exponent. The exponents can also be converted into IEC prefixes. It is a bit of a strange formatting. The numbers aren't actually expressed in binary per-se. They are expressed with scientific notation but with the base being 2 instead of 10. I imagine this may be useful for computer science applications. The inclusion of this binary formatting mode was motivated by the [prefixed package](https://pypi.org/project/prefixed/) which includes that sort of formatting.

In short I would like to drop support for these formatting modes. The codebase is already big and a bit cluttered, and the API has very many options. From a maintenance and documentation perspective it would be nice to slim things down a bit. Also value/uncertainty formatting currently isn't supported for binary formatting since I haven't had interest in writing tests and getting it working.

This issue would obviate #101 and #109.
